lib/TWCManager/TWCMaster.py

```diff
@@ -293,6 +293,10 @@ def getScheduledAmpsTimeFlex(self):
             slave = next(iter(self.slaveTWCs.values()))
             vehicle = slave.getLastVehicle()
             if vehicle != None:
+                # we update current SOC - but only when charging 
+                if slave.isCharging==1:
+                    vehicle.update_charge()
```

Comment:
```suggestion
                            self.queue_background_task({"cmd": "checkCharge"})
```
Rather than making a blocking API call in-line here, you might want to queue an instance of the `checkCharge` background task, so that the charge will get updated in the background by the next time this code executes.